share/functions/man.fish

```diff
@@ -6,25 +6,23 @@ end
 function man --description "Format and display the on-line manual pages"
     # Work around the "builtin" manpage that everything symlinks to,
     # by prepending our fish datadir to man. This also ensures that man gives fish's
-    # man pages priority, without having to put fish's bin directories first in $PATH
+    # man pages priority, without having to put fish's bin directories first in $PATH.
 
+    # Preserve the existing MANPATH, and default to the system path (the empty string).
     set -l manpath
     if set -q MANPATH
         set manpath $MANPATH
-    else if command -qs manpath
-        set manpath (command manpath)
```

Comment:
This removes the `command manpath` call entirely, which I don't like. There are some annoying man implementations out there.

Unless you have tested this on all of them, I would not merge this.

Reply:
It works as expected on `man-db`, FreeBSD man, OpenBSD man, and Darwin man. Admittedly, I don't have any commercial *nix variants to test, not have I tested Illumos.

Reply:
Faso: Man should have this behavior in all cases where there is an empty MANPATH or MANPATH has an empty component (like ends in ":")

Reply:
Turns out POSIX explicitly does not standardize $MANPATH (!). Please leave this in, in case any weird systems out there need it.

This function is completely performance-noncritical.

Reply:
Yeah, POSIX doesn't specify here, but as far as I can tell, any AT&T-descended Unix-like implementation of `manpath(1)` should work as expected (not unlike the convention of `$PATH` with an empty entry signifying PWD).

Still, if you feel strongly about it, I can put the call back. It just feels better to omit it to me, as adding an extra colon to `$MANPATH` has been the standard method for extending the variable as long as I can remember.

Reply:
Okay, that one tries both `manpath` and `man --path` (Haiku!), but it needs to know the actual value.

If neither of those exist, it needs to read /etc/man.conf, because mandoc does not provide `manpath` or `man --path`...

So:

- For Haiku, `man --path` needs to be used instead of `manpath`, so this wasn't working
- For mandoc, this didn't work because it provides no way of easily getting the default manpath
- For other things, the empty entry works

So: Yeah, the empty entry is probably alright. Go ahead.

Reply:
I think whoever wrote that code was not familiar with `manpath`. Because we're just trying to get fish's directories appended to the default paths and that's exactly what an empty path component will do.

```console
$ env MANPATH=/usr/local/share/fish/man manpath
/usr/local/share/fish/man
$ env MANPATH=/usr/local/share/fish/man: manpath
/usr/local/share/fish/man:/usr/local/share/man:/usr/share/man:/Applications/Xcode.app/Contents/Developer/usr/share/man:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man
$ env MANPATH=:/usr/local/share/fish/man manpath
/usr/local/share/man:/usr/share/man:/Applications/Xcode.app/Contents/Developer/usr/share/man:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man:/usr/local/share/fish/man
```

(And actually do you even need to run `manpath`(1) at all? Just setting MANPATH up with one of the components empty will let it find the fish man pages as well as ones in any of the default locations, man will internally have figured out the same thing you're seeing from `manpath` above.)

Reply:
On some systems `man` forks out to `manpath`, and on others it uses the same shared code in its own process. But as far as I know, on systems with man-db (read: Linux with GNU userspace), `manpath` is a bygone vestige that mostly exists to allow the user to check the fully resolved path used by `man`.

Reply:
Er, so I'm wrong about what modern `man` does with MANPATH? How does it handle a colon at the start or end?

Reply:
No, you're correct, I was just remarking on the actual use of the `manpath` utility. All versions of man I've ever used exhibit the colon behavior; however, some of them resolve the final manpath internally, while others call the `manpath` binary. One systems which do the former, the binary really just seems to be there for historical and testing purposes.